test-network-function/lifecycle/suite.go

```diff
@@ -104,9 +108,65 @@ var _ = ginkgo.Describe(common.LifecycleTestKey, func() {
 		}
 
 		testOwner(&configData)
+
+		testScaling(&configData)
 	}
 })
 
+func waitForAllDeploymentsReady(namespace string, timeout, pollingPeriod time.Duration) {
+	gomega.Eventually(func() []string {
+		_, notReadyDeployments := getDeployments(namespace)
+		return notReadyDeployments
+	}, timeout, pollingPeriod).Should(gomega.HaveLen(0))
+}
+
+func testScaling(configData *common.ConfigurationData) {
+	ginkgo.It("Testing deployment scaling", func() {
+		defer results.RecordResult(identifiers.TestScalingIdentifier)
+
+		// Map to register the deployments that have been already tested
+		deploymentNames := make(map[string]bool)
+
+		for _, cut := range configData.ContainersUnderTest {
+			namespace := cut.Oc.GetPodNamespace()
+
+			// Get deployment name and check whether it was already tested.
+			// ToDo: Proper way (helper/handler) to do this.
+			podNameParts := strings.Split(cut.Oc.GetPodName(), "-")
+			deploymentName := podNameParts[0]
+
+			if _, alreadyTested := deploymentNames[deploymentName]; alreadyTested {
+				continue
+			}
+
+			// Get deployment's current replicaCount
+			deployments, _ := getDeployments(namespace)
+			replicaCount := deployments[deploymentName].Replicas
+
+			// ScaleIn, removing one pod from the replicaCount
+			scaleInHandler := scaling.NewScaling(common.DefaultTimeout, namespace, deploymentName, (replicaCount - 1))
+			testScaleIn, errScaleIn := tnf.NewTest(common.GetContext().GetExpecter(), scaleInHandler, []reel.Handler{scaleInHandler}, common.GetContext().GetErrorChannel())
```

Comment:
``errScaleIn -> err `` for simplicity, err is used in the next two lines only

Reply:
Done.